### PR TITLE
Add Handheld Electronic Game.dat

### PR DESCRIPTION
**Pull request by lasers**
Related: https://github.com/libretro/libretro-database/issues/330
Closes https://github.com/libretro/libretro-database/issues/330.

### CODE REVIEW

**Comment by RobLoach** [comment]
What is this? Should likely follow No-Intro standards, don't need the company name.

**Comment by lasers** [comment]
A replacement for `Nintendo - Game and Watch.dat`. This style will keep all manufacturers together first, then games... making it possible to view a list of Nintendo games... or a list of VTech games... etc.

File: dat/Handheld Electronic Game.dat [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+game (
+	name "(Nintendo) Donkey Kong Circus (Panorama Screen)"
+	description "(Nintendo) Donkey Kong Circus (Panorama Screen)"
+	rom ( name "(Nintendo) Donkey Kong Circus (Panorama Screen).mgw" size 1450111 crc 4c61471c md5 a2dc6900bc58aab4090ceb1cce5bf612 sha1 e051de6d15fa89522557e667e2f344a05f27d05e )
+)
+ 
+game (
+	name "(Nintendo) Donkey Kong II (Multi Screen)"
+	description "Donkey Kong II (Multi Screen)"
+	rom ( name "(Nintendo) Donkey Kong II (Multi Screen).mgw" size 723767 crc 0bdc080b md5 29c7e0211a7033a11ab63ba3f7fe7c98 sha1 2848b2c7309c3c062e1aed3439d85b3fa1ea38ea )
```
Comment (RobLoach) [inline dat/Handheld Electronic Game.dat]: Could likely be arranged with.....

```
game (
 	name "Donkey Kong II (Multi Screen)"
 	description "Donkey Kong II (Multi Screen)"
        developer "Nintendo"
 	rom ( name "Donkey Kong II (Multi Screen).mgw" size 723767 crc 0bdc080b md5 29c7e0211a7033a11ab63ba3f7fe7c98 sha1 2848b2c7309c3c062e1aed3439d85b3fa1ea38ea )
)
```

Comment (lasers) [inline dat/Handheld Electronic Game.dat]: 56 total. 18 Nintendo. 25 VTech. 1 Bandai. 2 Coleco. 3 Elektronika. 1 Epoch. 1 Gakken. 3 Mattel Electronics. 1 Queentex. 2 Tomytronic. 1 Unknown.

**Comment by lasers** [comment]
@RobLoach: Look what else snuck in here. Merge dis pls.

**Comment by lasers** [comment]
I think during that conversation where he was okay with it... I looked into the games in the `Game and Watch.dat` to see that more than half games are non-Nintendo.

I also found a README... or he supplied a source link where the games came from. http://www.madrigaldesign.it/sim/download.php 

In that link, you can see Nintendo is not the only manufacturer. My original goal was to work on `Game and Watch` to make it more accurate... `Nintendo - Game & Watch` only to find that it had like 12 or 16 games... VTech got 36 games, I think.. And there were many other small manufacturers.

I did start splitting them up... but I don't know if guys would want `Nintendo - Game and Watch`, `VTech - Time and Fun`, `Coleco - Handheld Electronic Game`, `Bandai - Handheld Electronic Game`, etc... all over RetroArch's sorted XMB playlists. They don't actually have system name... Just brand... but I stopped anyway. Here's one for `Electronika - Handheld Electronic Game`. https://github.com/lasers/libretro-database/commit/c4ad801e1619fd8de7d024a5fee0d95e9730b904 

For that, we shouldn't use Nintendo's `Game and Watch` name/brand/logo to represent non-Nintendo games. I'm also wondering about Mr. Game and Watch (the dude) in the assets.... Dunno what we should replace it... only if we decide on replacing it. One wouldn't want to use Mario logo on Sega Genesis games. 

Here's the conversation again (if you hadn't looked yet).  https://github.com/libretro/libretro-database/issues/330 

--- 

This playlist is different than others because it's the only one with merged manufacturers. If I put NES and SNES together, I would start with `NES -` and another one with `SNES - ` to keep them separated in a merged list. What will `developer` entries do? 

Bunch of small individual handheld with different manufacturers, different games and different styles all in one playlist. I organized it what I can and not just going with ABC's.

**Comment by lasers** [comment]
@RobLoach Let's look at this. I'm so over with G&W. If you wanted to strip the (series) off too, let me know... And if you approve this, I could compile and add rdb too? Thanks.

**Comment by twinaphex** [comment]
@RobLoach Let me know if this should be merged, thanks.

**Comment by lasers** [comment]
This breaks `Handheld Electronic Game-->Information` because I used `manufacturer` (found and used in most DAT tools) instead of `developer` although they're manufacturers. The fix could be done in `menu/menu_displaylist.c` to correct this if desired.

I'll change everything to `developer`.

**Comment by lasers** [comment]
@RobLoach I think for most part, this database is almost same as the current `Game and Watch.dat` but with series included to help differentiate only some games. The sooner you merge this, the sooner I can test the new changes to make sure everything works okay and move on.

**Comment by RobLoach** [comment]
Seems good with me! I'll start the renaming process in the other projects.

**Comment by lasers** [comment]
@RobLoach Long time! By the way, I found couple of outdated mgw (different hashes) as I noticed that some games didn't show up in the playlist... And leiradel did make a new one today for Donkey Kong today. I unleashed hell on him with all those new issues in gw-libretro so I think we may get some new hashes in the near future (or not that near future).

I just need to know which you prefer.
```
(Nintendo) GameName (Series)
GameName (Nintendo, Series)
GameName (Series)  <-- This one, I think... are what we're doing here.
GameName
```
It's hard to differentiate games without series, but I think we can live without manufacturer. I think the old G&W one was just `GameName`. I also made thumbnails today. >_>

I will recheck the hashes and make minor updates.

**Comment by RobLoach** [comment]
`GameName (Series)`

Matches the No-Intro pattern the closest. Could even eventually move the Series name into a `franchise` key in the DAT, or similar.

**Pull request merged**